jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitProjectWizardTest.kt

```diff
@@ -0,0 +1,106 @@
+        try {
+            createProject { step ->
+                if (step is ProjectTypeStep) {
+                    assertTrue(step.setSelectedTemplate("SAM", null))
+                    val builder = myWizard.projectBuilder as SamInitModuleBuilder
+                    builder.runtimeSelectionPanel.samExecutableField.text = ""
+                }
+            }
+            fail("Exception was not thrown")
+        } catch (e: RuntimeException) {
+            // expected a runtime exception
+            assertTrue(e.message == "SAM CLI executable not configured" ||
+                e.message == "com.intellij.ide.projectWizard.ProjectTypeStep is not validated")
+        }
+    }
+
+    fun testPython36Runtime() {
+        helloWorldTest(Runtime.PYTHON3_6, PythonSdkType::class)
+    }
+
+    fun testPython27Runtime() {
+        helloWorldTest(Runtime.PYTHON2_7, PythonSdkType::class)
+    }
+
+    fun testJavaRuntime() {
+        helloWorldTest(Runtime.JAVA8, JavaSdk::class)
+    }
+
+    fun helloWorldTest(runtime: Runtime, sdkType: KClass<out SdkType>) {
+        SamSettings.getInstance().savedExecutablePath = "sam"
+
+        createProject { step ->
+            val stepNum = myWizard.currentStep
+            when (step) {
+                is ProjectTypeStep -> { // Wizard first page
+                    assertEquals(0, stepNum)
+                    assertTrue(step.setSelectedTemplate("SAM", null))
+                    // step count changes after we select SAM
+                    assertEquals("Found steps: ${myWizard.sequence.selectedSteps}.", 4, myWizard.sequence.selectedSteps.size)
+
+                    // builder is not of type until we select SAM
+                    val builder = myWizard.projectBuilder as SamInitModuleBuilder
+                    // ensure we have set the executable path
+                    assertNotNull(builder.runtimeSelectionPanel.samExecutableField.text)
+                    // select python
+                    builder.runtimeSelectionPanel.runtime.selectedItem = runtime
+                }
+                is ChooseTemplateStep -> { // Invisible to user, but forced step
+                    assertEquals(1, stepNum)
+                }
+                is SamInitTemplateSelectionStep -> { // Custom template selection step
+                    assertEquals(2, stepNum)
+
+                    val builder = myWizard.projectBuilder as SamInitModuleBuilder
+                    assertEquals(runtime, builder.runtimeSelectionPanel.runtime.selectedItem)
+                    assertInstanceOf(builder.getSdkType(), sdkType.java)
+                    assertEquals("SAM Hello World", step.templateSelectionPanel.selectedTemplate!!.name)
+                }
+                is ProjectSettingsStep -> {
+                    assertEquals(3, stepNum)
+                }
+                else -> {
+                    fail("Unknown step!")
+                }
+            }
+        }
+    }
+
+    override fun setUp() {
```

Comment:
Can we move this to the beginning of the test class? I'd be confused as a reviewer seeing the `setUp` at the end

Reply:
done